gloo/transport/ibverbs/pair.cc

```diff
@@ -478,43 +607,67 @@ void Pair::handleCompletion(struct ibv_wc* wc) {
         ": ",
         ibv_wc_status_str(wc->status));
 
-    GLOO_ENFORCE_GT(mappedSendRegions_.size(), 0);
-    GLOO_ENFORCE_EQ(mappedSendRegions_.count(slot), 1);
-    mappedSendRegions_.erase(slot);
+    mappedSendRegions_[slot].pop_front();
   } else {
     GLOO_ENFORCE(false, "Unexpected completion with opcode: ", wc->opcode);
   }
 }
 
 void Pair::send(Buffer* buffer, size_t offset, size_t length, size_t roffset) {
```

Comment:
another n00b question, why there are two send? Is it possible that we can have comments to explain under what condition this send will be used? For example the comment of the first one says "// Send from the specified buffer to remote side of pair." I guess this one is similar. Even the code looks similar.

Reply:
It's for the two different types of buffers. Buffer is bound to a specific pair and reuses the memory region where as UnboundBuffer can be used across multiple pairs and will bind the memory region just for that transfer.

I tried to generalize this logic but had to split it given the differences in apis between the two

Reply:
Got it, thanks for your explanation.